### PR TITLE
Update notification panel content

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue

#223 

## Overview
Updated the content of the `NotificationPanel` to be be more specific so that it matches the design. The `NotificationPanel` now accepts a collection of notifications as to be presented as a list.

## Reason
Currently the `NotificationPanel` takes arbitrary content but this needs constraining further.

## Work carried out
- [x] Add notification list component with list item
- [x] Integrate changes
- [x] Update developer documentation
- [ ] Update design documentation

## Screenshot
Screenshot 2019-11-07 at 16 27 59

## Developer notes
- There is currently a bug in the `Sidebar` which means the notifications are not displayed
- Use the `Masthead` `With navigation` Storybook story for an example
- The `rn-notifications-item__indicator` could be reused instead of [a SVG circle](https://github.com/Royal-Navy/standards-toolkit/blob/24fb581c99324d4fb2b74f02c1721dff4766fb3b/packages/react-component-library/src/fragments/NotificationPanel/index.tsx#L119). This will be done in a follow up PR

### CODE REVIEW

**Comment by hxltrhuxze** [comment]
Looking great!

Just a couple of styling tweaks:
- Can we change the title to have `line-height: 1.3`? I realise it's set to `~1.2` in the Sketch document, however it looks more cramped in the browser (most likely due to subtle differences in font rendering between Sketch and CSS)
- When you hover on an item, can we make the Avatar and the white items (Name & Review) turn blue? At the moment, only the Avatar changes when it is hovered on which can be jarring as it implies clicking the avatar does something different to clicking the notification item. The Notification item by itself doesn't have a hover state so it's tricky trying to work out where you should click.
- When hovering on the "View all notifications" link, can we remove the underline and instead change the colour of the link & the right chevron to blue `color(primary, 300)`? And then can we also animate the chevron to the right by `~4px`?

File: packages/docs-site/src/library/pages/components/notifications.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+---
+title: Notifications
+description: todo
```
Comment (m7kvqbe1) [inline packages/docs-site/src/library/pages/components/notifications.md]: This needs populating still.

File: packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+}) => {
+  const notificationArrowPosition =
+    notificationPlacement === 'right' ? 'left_bottom' : 'top_right'
+
+  const [showNotifications, setShowNotifications] = useState(false)
+  const [notificationPosition, setNotificationPosition] = useState()
+  const node = useRef()
+
+  function documentClick(event: Event) {
+    // workaround for undefined error in typescript
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx:59]: Anyone know what this is about? Is this to deal with browser APIs missing in node when doing SSR?

File: packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    if (current.contains(event.target)) {
+      return
+    }
+
+    setShowNotifications(false)
+  }
+
+  function toggleNotifications(event: React.SyntheticEvent) {
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx:71]: I know this isn't your code but maybe we could refactor some of this into a custom hook?

**Review by m7kvqbe1 — APPROVED** [review]
Looks good - just a couple of comments - mainly about some of the pre-existing code.

File: packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx (hunk 3 above)
Comment (thyhjwb6) [reply packages/react-component-library/src/fragments/NotificationPanel/NotificationPanel.tsx:71]: I think that is a good idea which I would like to do but would prefer to keep this change focused and do a follow up PR.

**Pull request merged**